File: scripts/generate-pwa-icons.js
#!/usr/bin/env node

/**
 * Script para gerar ícones PWA a partir do favicon.png
 * Requer: npm install sharp
 */

const sharp = require('sharp');
const path = require('path');
const fs = require('fs');

const faviconPath = path.join(__dirname, '..', 'client', 'public', 'favicon.png');
const outputDir = path.join(__dirname, '..', 'client', 'public');

// Dimensões necessárias para PWA
const sizes = [192, 512];

async function generateIcons() {
  try {
    // Verificar se o favicon existe
    if (!fs.existsSync(faviconPath)) {
      console.error(`❌ Erro: Arquivo não encontrado: ${faviconPath}`);
      process.exit(1);
    }

    console.log(`✓ Favicon carregado: ${faviconPath}`);

    // Gerar ícones para cada tamanho
    for (const size of sizes) {
      // Ícone padrão
      await sharp(faviconPath)
        .resize(size, size, {
          fit: 'contain',
          background: { r: 255, g: 255, b: 255, alpha: 0 }
        })
        .png()
        .toFile(path.join(outputDir, `icon-${size}.png`));
      console.log(`✓ Criado: icon-${size}.png`);

      // Ícone maskable (com espaço ao redor)
      const innerSize = Math.floor(size * 0.8);
      const offset = (size - innerSize) / 2;

      await sharp(faviconPath)
        .resize(innerSize, innerSize, {
          fit: 'contain',
          background: { r: 255, g: 255, b: 255, alpha: 0 }
        })
        .extend({
          top: Math.floor(offset),
          bottom: Math.floor(offset),
          left: Math.floor(offset),
          right: Math.ceil(offset),
          background: { r: 255, g: 255, b: 255, alpha: 0 }
        })
        .png()
        .toFile(path.join(outputDir, `icon-maskable-${size}.png`));
      console.log(`✓ Criado: icon-maskable-${size}.png`);
    }

    console.log('\n✅ Ícones PWA gerados com sucesso!');
    console.log('\nOs seguintes ícones foram criados em client/public/:');
    for (const size of sizes) {
      console.log(`  - icon-${size}.png`);
      console.log(`  - icon-maskable-${size}.png`);
    }

  } catch (error) {
    console.error(`❌ Erro ao gerar ícones: ${error.message}`);
    process.exit(1);
  }
}

generateIcons();

File: scripts/generate-pwa-icons.py
#!/usr/bin/env python3
"""
Script para gerar ícones PWA a partir do favicon.png
Requer: pip install Pillow
"""

import os
from PIL import Image

# Caminhos
favicon_path = os.path.join(
    os.path.dirname(__file__),
    '..',
    'client',
    'public',
    'favicon.png'
)

output_dir = os.path.join(
    os.path.dirname(__file__),
    '..',
    'client',
    'public'
)

# Dimensões necessárias para PWA
sizes = [192, 512]

try:
    # Abrir a imagem
    img = Image.open(favicon_path)
    
    # Converter para RGBA se necessário (para suportar transparência)
    if img.mode != 'RGBA':
        img = img.convert('RGBA')
    
    print(f"✓ Favicon carregado: {favicon_path}")
    print(f"  Tamanho original: {img.size}")
    
    # Gerar ícones para cada tamanho
    for size in sizes:
        # Ícone padrão
        icon = img.resize((size, size), Image.Resampling.LANCZOS)
        icon_path = os.path.join(output_dir, f'icon-{size}.png')
        icon.save(icon_path, 'PNG')
        print(f"✓ Criado: {os.path.basename(icon_path)}")
        
        # Ícone maskable (com espaço ao redor para garantir visibilidade)
        # Criar um fundo branco e colocar a imagem no centro
        maskable = Image.new('RGBA', (size, size), (255, 255, 255, 0))
        
        # Redimensionar a imagem um pouco menor para deixar espaço
        inner_size = int(size * 0.8)
        inner_icon = img.resize((inner_size, inner_size), Image.Resampling.LANCZOS)
        
        # Calcular posição para centralizar
        offset = (size - inner_size) // 2
        maskable.paste(inner_icon, (offset, offset), inner_icon)
        
        maskable_path = os.path.join(output_dir, f'icon-maskable-{size}.png')
        maskable.save(maskable_path, 'PNG')
        print(f"✓ Criado: {os.path.basename(maskable_path)}")
    
    print("\n✅ Ícones PWA gerados com sucesso!")
    print("\nOs seguintes ícones foram criados em client/public/:")
    for size in sizes:
        print(f"  - icon-{size}.png")
        print(f"  - icon-maskable-{size}.png")
    
except FileNotFoundError:
    print(f"❌ Erro: Arquivo não encontrado: {favicon_path}")
    print("Certifique-se de que favicon.png existe em client/public/")
except ImportError:
    print("❌ Erro: Pillow não está instalado")
    print("Instale com: pip install Pillow")
except Exception as e:
    print(f"❌ Erro ao gerar ícones: {e}")
